src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp

```diff
@@ -1269,29 +767,45 @@ class at_naxxramas_hub_portal : public AreaTriggerScript
 
     bool OnTrigger(Player* player, AreaTrigger const* /*trigger*/) override
     {
-        if (player->IsAlive() && !player->IsInCombat())
-        {
-            if (InstanceScript *instance = player->GetInstanceScript())
-            {
-                bool AreAllWingsCleared = instance->GetBossState(BOSS_MAEXXNA) == DONE
-                    && (instance->GetBossState(BOSS_LOATHEB) == DONE)
-                    && (instance->GetBossState(BOSS_THADDIUS) == DONE)
-                    && (instance->GetBossState(BOSS_HORSEMAN) == DONE);
+        if (!player->IsAlive() || player->IsInCombat())
+            return false;
 
-                if (AreAllWingsCleared)
-                {
-                    player->TeleportTo(533, sapphironEntryTP.m_positionX, sapphironEntryTP.m_positionY, sapphironEntryTP.m_positionZ, sapphironEntryTP.m_orientation);
-                    return true;
-                }
-            }
-        }
-        return false;
+        InstanceScript* instance = player->GetInstanceScript();
+        if (!instance)
+            return false;
+
+        if ((instance->GetBossState(BOSS_MAEXXNA)  != DONE) ||
+            (instance->GetBossState(BOSS_LOATHEB)  != DONE) ||
+            (instance->GetBossState(BOSS_THADDIUS) != DONE) ||
+            (instance->GetBossState(BOSS_HORSEMAN) != DONE))
+            return false;
+
+        player->TeleportTo(SapphironTeleportPos);
+        return true;
     }
 };
 
+void AddSC_boss_anubrekhan();
+void AddSC_boss_maexxna();
+void AddSC_boss_patchwerk();
+void AddSC_boss_grobbulus();
+void AddSC_boss_razuvious();
+void AddSC_boss_kelthuzad();
+void AddSC_boss_loatheb();
+void AddSC_boss_noth();
+void AddSC_boss_gluth();
+void AddSC_boss_sapphiron();
+void AddSC_boss_four_horsemen();
+void AddSC_boss_faerlina();
+void AddSC_boss_heigan();
+void AddSC_boss_gothik();
+void AddSC_boss_thaddius();
+
 void AddSC_instance_naxxramas()
 {
-    new instance_naxxramas();
-    new boss_naxxramas_misc();
+    RegisterInstanceScript(instance_naxxramas, NaxxramasMapId);
```

Comment:
```suggestion
    RegisterInstanceScript(NaxxramasScriptName, NaxxramasMapId);
```

It is already defined in naxxramas.h